bin-src/lib/getDependentStoryFiles.ts

```diff
@@ -153,7 +217,11 @@ export async function getDependentStoryFiles(
     return mod.modules?.length ? mod.modules.map((m) => normalize(m.name)) : [normalize(mod.name)];
   }
 
-  const tracedFiles = changedFiles.filter(untrace);
+  // append the third-party modules (respecting any untraced rules)
+  const tracedFiles = changedModules
+    .reduce((acc, mod) => acc.concat(thirdPartyModules[mod]), changedFiles)
+    .filter(untrace);
```

Comment:
```suggestion
  const tracedFiles = [...changedModules.flatMap(mod => thirdPartyModules[mod]), ...changedFiles]
    .filter(untrace);
```

Might be a bit simpler?